lib/streamlit/type_util.py

```diff
@@ -578,19 +616,47 @@ def convert_anything_to_df(
             return pd.DataFrame([])
         return pd.DataFrame(data)
 
-    if (
-        is_type(data, _SNOWPARK_DF_TYPE_STR)
-        or is_type(data, _SNOWPARK_TABLE_TYPE_STR)
-        or is_type(data, _PYSPARK_DF_TYPE_STR)
-    ):
-        if is_type(data, _PYSPARK_DF_TYPE_STR):
-            data = data.limit(max_unevaluated_rows).toPandas()
-        else:
-            data = pd.DataFrame(data.take(max_unevaluated_rows))
+    if is_snowpark_data_object(data):
+        data = data.limit(max_unevaluated_rows).to_pandas()
+        if data.shape[0] == max_unevaluated_rows:
+            st.caption(
+                f"⚠️ Showing only {string_util.simplify_number(max_unevaluated_rows)} rows. "
+                "Call `to_pandas()` on the dataframe to show more."
+            )
+        return cast(pd.DataFrame, data)
```

Comment:
nit: it looks like the overall pattern with warning etc. is repeated - might be worth to look into whether there is a good refactoring possibility here

Reply:
Yeah, all of this will be somewhat refactored next quarter with the support more dataframe formats project. A bit tricky here is that while the warning are mostly the same, they are still fully dependent on the underlying data structure (e.g. whats the method to actually convert it to pandas.